azkaban-common/src/main/java/azkaban/executor/ExecutableFlow.java

```diff
@@ -475,4 +482,53 @@ public VersionSet getVersionSet() {
   public void setVersionSet(final VersionSet versionSet) {
     this.versionSet = versionSet;
   }
+
+  /**
+   * Getter of flattened flow properties and flow params. This API takes lazy
+   * loading approach. If the properties are not set, then they are first set.
+   * @return Returns the flattened flow props overridden by flow params.
+   */
+  public Props getFlowPropsAndParams(final ProjectLoader projectLoader) {
+    // If the object is set, just return it.
+    if (this.flattenedFlowPropsAndParams != null) {
+      return this.flattenedFlowPropsAndParams;
+    }
+
+    Props props = new Props();
+    // Fetch the flow properties
+    if (FlowLoaderUtils.isAzkabanFlowVersion20(this.azkabanFlowVersion)) {
+      props = FlowLoaderUtils.loadPropsFromYamlFile(projectLoader,
+          this, null);
+    } else {
+      final Map<String, Props> propsMap = projectLoader.
+          fetchProjectProperties(projectId, version);
+      if (null != propsMap) {
+        Props finalProps = props;
```

Comment:
finalProps object reference is not needed. You can directly do putAll on props.

Reply:
true. remnant of lambda code, will clean it up.